### PR TITLE
Implement qm interface

### PR DESCRIPTION
**Pull request by Kobo-coder**
- Implemented first iteration of the QualityModel interface.
- Made changes to the interface in the process -> Renamed and removed/changed as discussed.
- Ran the Python BLACK formatter

There are no tests yet, but comments on code would be appreciated.

### CODE REVIEW

File: byoqm/qualitymodel/qualitymodel.py [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
-        childen: List[Characteristic],
-        aggregations: Callable[..., int | float],
-    ):
+        child: Characteristic | Metric
+        ):
         """Inserts a new characteristic into the quality model.
 
-        Assumes that `aggregations` is a function with n parameters, where n is the
-        number of children.
+        Aggregation is a function with the signature: f(int | float list, int | float list) -> int | float
```
Comment (toffernator) [inline byoqm/qualitymodel/qualitymodel.py:25]: insert accepts no `aggregation` anymore

File: byoqm/qualitymodel/treequalitymodel.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def measure(self):
+        measurement = None
+        for c in self.children:
+            measurement += c.measure()
+        return measurement
+
+
+class TreeQualityModel(QualityModel):
+    def __init__(self):
+        self.root = Node("root", None, [])
```
Comment (toffernator) [inline byoqm/qualitymodel/treequalitymodel.py:23]: IMO making the root's name "quality" makes more sense. What are people's thoughts?

File: .fleet/settings.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "editor.guides": [],
+    "excluded": [],
+    "toolchains": []
```
Comment (toffernator) [inline .fleet/settings.json:4]: Looks like a pretty dead config to me, if it did stuff I could see the point of including it.

**Review by toffernator — COMMENTED** [review]
Looks good, just some minor concerns.

File: byoqm/qualitymodel/treequalitymodel.py (hunk 2 above)
Comment (FrederikRothe) [reply byoqm/qualitymodel/treequalitymodel.py:23]: Could do that yeah, that would go nicely with our overall abstraction and drawing of the Quality Model Tree.

File: .fleet/settings.json (hunk 3 above)
Comment (FrederikRothe) [reply .fleet/settings.json:4]: x   x
   o

very ded yeah.. cut it

**Review by FrederikRothe — APPROVED** [review]
Looks good, with respect to Toffer's above mentioned details.

File: .fleet/settings.json (hunk 3 above)
Comment (Kobo-coder) [reply .fleet/settings.json:4]: Accidently included it when recursively adding files. Will remove it.

File: byoqm/qualitymodel/treequalitymodel.py (hunk 2 above)
Comment (Kobo-coder) [reply byoqm/qualitymodel/treequalitymodel.py:23]: Will change it.

File: byoqm/qualitymodel/qualitymodel.py (hunk 1 above)
Comment (Kobo-coder) [reply byoqm/qualitymodel/qualitymodel.py:25]: 🫡

**Review by SadCheeto — APPROVED** [review]

**Pull request merged**